src/deepymod/model/deepmod.py

```diff
@@ -38,143 +37,150 @@ def forward(self, input: Tuple[TensorList, TensorList]) -> Tuple[TensorList, Ten
         sparse_thetas = self.apply_mask(thetas)
         self.coeff_vectors = self.calculate_coeffs(sparse_thetas, time_derivs)
 
-
     def apply_mask(self, thetas: TensorList) -> TensorList:
-        """ Function that applies the sparsity mask to the library function.  
+        """ Applies the sparsity mask to the feature (library) matrix.
 
         Args:
-            thetas (TensorList): List of library functions, one for every output. 
+            thetas (TensorList): List of all library matrices of size [(n_samples, n_features) x n_outputs].
 
         Returns:
-            TensorList: The sparse version of the library function. 
+            TensorList: The sparse version of the library matrices of size [(n_samples, n_active_features) x n_outputs].
         """
         sparse_thetas = [theta[:, sparsity_mask] for theta, sparsity_mask in zip(thetas, self.sparsity_masks)]
         return sparse_thetas
 
     @abstractmethod
     def calculate_coeffs(self, sparse_thetas: TensorList, time_derivs: TensorList) -> TensorList:
-        """Abstract method to compute the coefficients for the library.
+        """Abstract method. Specific method should return the coefficients as calculated from the sparse feature
+        matrices and temporal derivatives.
+
         Args:
-            sparse_thetas (TensorList): List of sparsified library functions, one for every output. 
-            time_derivs (TensorList): List of time derivatives
+            sparse_thetas (TensorList): List containing the sparse feature tensors of size (n_samples, n_active_features).
+            time_derivs (TensorList): List containing the time derivatives of size (n_samples, n_outputs).
 
         Returns:
-            Coefficients (TensorList): Return the coefficients.
+            (TensorList): Calculated coefficients of size (n_features, n_outputs).
         """
         pass
 
 
 class Estimator(nn.Module,  metaclass=ABCMeta):
-    """Abstract class implementing the sparsity estimator set to the function approximator. 
-
-    Args:
-        nn (PyTorch Class): Module of the function approximator, typically a neural network. 
-    """
     def __init__(self) -> None:
+        """Abstract baseclass for the sparse estimator module."""
         super().__init__()
         self.coeff_vectors = None
 
     def forward(self, thetas: TensorList, time_derivs: TensorList) -> TensorList:
-        """This function nomalized the library and time derivatives and calculates the corresponding sparisity mask.  
+        """The forward pass of the sparse estimator module first normalizes the library matrices
+        and time derivatives by dividing them by their l2 norm, than calculate the coefficient vectors
```

Comment:
Here, I would be more precise since there are many ways of taking the norm of a matrix: l2 norm column wise (which means each covariate vector has a norm one if I am not mistaken?)

Reply:
Good idea, I fixed it!